admin/jobs/river/validate_deployments.go

```diff
@@ -44,14 +53,20 @@ func (w *ValidateDeploymentsWorker) validateDeployments(ctx context.Context) err
 			afterName = projs[len(projs)-1].Name
 		}
 
-		// Process batch
+		// Process batch concurrently
```

Comment:
Not sure if this really matters, but a "problem" here is that if one or a few upgrades are slow, it holds up the entire batch. So potentially a lot of the time may be spent waiting for just a few slow upgrades.

If you want to ensure that it consistently runs `n` concurrent upgrades, you can consider
1. Using `sync.WaitGroup` to start `n` concurrent workers that consume projects to upgrade from a shared channel. When the channel is closed, it can return.
2. Having a separate goroutine that pages through `FindProjects`, sending each project to the worker's shared channel. Since channels are blocking, it will automatically never prefetch more projects than the page size.

Reply:
Updated to a shared channel and a worker pool.